app/views/store/homepage.html.erb

```diff
@@ -3,9 +3,20 @@
     <%= current_user.first_name %>
   <% end %>
 </h1>
-
+</br>
+<%= link_to 'View Previous Deals', past_deals_path %>
 </br>
 <% @live_deals&.each do |deal| %>
+  <%= render deal %>
+  <% if deal.quantity.zero? %>
+    <button disabled>Sold Out</button>
+  <% else %>
+    <%= button_to 'Buy', line_items_path(deal_id: deal), id: "deal_button_#{deal.id}" %>
```

Comment:
what is the need of giving id here?

Reply:
to provide button_id which is required when action cable runs js file where button is found by using document.getElementById and is disabled when a deal is sold out